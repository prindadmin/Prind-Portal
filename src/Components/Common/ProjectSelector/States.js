export const QUIESCENT = 'QUIESCENT'
export const LOADING = 'LOADING'
export const LOADING_ERROR = 'LOADING_ERROR'
export const UPDATE_ERROR = 'UPDATE_ERROR'
export const PROJECT_TYPE_SELECTOR_OPEN = 'PROJECT_TYPE_SELECTOR_OPEN'
