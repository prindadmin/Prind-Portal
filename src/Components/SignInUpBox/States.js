export const QUIESCENT = 'QUIESCENT'
export const LOADING = 'LOADING'
export const PASSWORD_RESET_SUCCESS = 'PASSWORD_RESET_SUCCESS'
export const PASSWORD_RESET_FAILED = 'PASSWORD_RESET_FAILED'
