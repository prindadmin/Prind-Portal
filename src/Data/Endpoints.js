export const DEFAULTPAGE = ""
export const DEFAULTLOGGEDINPAGE = "/inception"
export const PROJECTDETAILSPAGE = "/project"
export const SIGNINPAGE = "/signin"
export const SIGNUPPAGE = "/signup"
export const MEMBERPAGE = "/team"
