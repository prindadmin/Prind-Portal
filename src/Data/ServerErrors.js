
export const PROJECT_NOT_FOUND = "PROJECT_NOT_FOUND"
export const INSUFFICIENT_PERMISSION = "INSUFFICIENT_PERMISSION"
