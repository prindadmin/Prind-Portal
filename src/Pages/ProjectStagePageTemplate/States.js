export const QUIESCENT = 'QUIESCENT'

export const CONTENT_FETCH_IN_PROGRESS = 'CONTENT_FETCH_IN_PROGRESS'
export const CONTENT_FETCH_SUCCESSFUL = 'CONTENT_FETCH_SUCCESSFUL'
export const CONTENT_FETCH_FAILED = 'CONTENT_FETCH_FAILED'
