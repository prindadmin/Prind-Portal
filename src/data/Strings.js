
export const YES = "Yes"
export const NO = "No"

export const IF_YES_PROVIDE_DETAILS_BELOW = "If yes, please provide extra details below"
export const IF_XXX_PROVIDE_DETAILS_BELOW = "If XXX, please provide extra details below"
export const PLEASE_PROVIDE_DETAILS_HERE = "Please provide extra details here..."

export const PROJECT_LOADING = "Project loading, please wait..."
export const UNABLE_TO_FETCH_CONTENT = "Unable to load the content for this page.  Please try again.  If this problem persists, please contact technical support."
export const NO_PROJECT_SELECTED = "Please select a project"
export const CREATE_NEW_PROJECT = "Create a new project"
export const ERROR_CREATING_NEW_PROJECT = "There was an error creating a new project, please try again.  If the problem persists, please contact technical support."

export const PROJECT_NAME = "Project Name"
export const PROJECT_REFERENCE = "Project Reference"
export const PROJECT_ADDRESS = "Project Address"
export const FIELD_IS_REQUIRED = "(required)"

export const ADDRESS_LINE_1 = "Address Line 1"
export const ADDRESS_LINE_2 = "Address Line 2"
export const ADDRESS_LINE_3 = "Address Line 3"
export const ADDRESS_LINE_TOWN = "Town / City"
export const ADDRESS_LINE_REGION = "County"
export const ADDRESS_LINE_POSTAL_CODE = "Post Code"
export const ADDRESS_LINE_COUNTRY = "Country"

export const PROJECT_DESCRIPTION = "Project Description"
export const NO_PROJECT_DESCRIPTION = "No project description provided"

export const BUTTON_CREATE_PROJECT = "Create Project"
export const BUTTON_CANCEL = "Cancel"
export const BUTTON_SAVE_CHANGES = "Save Changes"

export const INCEPTION_PAGE_TITLE = "Project Inception"
export const INCEPTION_PAGE_DESCRIPTION = "Inception of a project begins with the idea. This idea is captured in a project brief."

export const FEASIBILITY_PAGE_TITLE = "Project Feasibility"
export const FEASIBILITY_PAGE_DESCRIPTION = "To confirm that the project brief is feasible it will be necessary to perform investigations which can be compiled into a feasibility study."

export const DESIGN_PAGE_TITLE = "Project Design"
export const DESIGN_PAGE_DESCRIPTION = "The Appointed design team will be collectively identifying hazards and considering mitagations for potential risks.  The assessment of risks will be complied into a single Design Risk Register."

export const TENDER_PAGE_TITLE = "Project Tender"
export const TENDER_PAGE_DESCRIPTION = "The design is complete, you are ready to build.  You will require a Principal Contractor.  It is the client's responsibility to provide Pre-Construction information."

export const CONSTRUCTION_PAGE_TITLE = "Project Construction"
export const CONSTRUCTION_PAGE_DESCRIPTION = "You have appointed a Principal contractor (PC).  The PC has full responsibility for Health and Safety during this step.  The PC will provide a Construction Phase Plan (CPP)."

export const HANDOVER_PAGE_TITLE = "Project Handover"
export const HANDOVER_PAGE_DESCRIPTION = "The building is complete.  The PC will have provided a Practical Completion Certificate."

export const OCCUPATION_PAGE_TITLE = "Project Occupation"
export const OCCUPATION_PAGE_DESCRIPTION = "The Principal Designer has finalised the Project Health and Safety File (HSF)."

export const REFURBISHMENT_PAGE_TITLE = "Project Refurbishment"
export const REFURBISHMENT_PAGE_DESCRIPTION = "Time for refurbishment? If your project will last more that 30 days, start new project.  Demolition? - look forward to helping you.\n\nStart new project"

export const EDIT_PROJECT_DETAILS = "Edit Project Details"

export const PROJECT_DIRECTORY_TITLE = "Project Directory"

export const PROJECT_DIRECTORY_ADD_NEW_TITLE = "Add New Project Member"

export const BUTTON_ADD_MEMBER_TO_PROJECT = "Add New Member"
export const ERROR_ADDING_MEMBER_TO_PROJECT = "There was an error adding member to the project, please try again.  If the problem persists, please contact technical support."
export const ERROR_REMOVING_MEMBER_FROM_PROJECT = "There was an error removing a member from the project, please try again.  If the problem persists, please contact technical support."

export const MEMBER_DETAILS = "Member Details"
export const MEMBER_FIRST_NAME = "First Name"
export const MEMBER_LAST_NAME = "Surname"
export const MEMBER_LANDLINE_PHONE_NUMBER_WORK = "Phone"
export const MEMBER_MOBILE_PHONE_NUMBER_WORK = "Mobile"
export const MEMBER_EMAIL_ADDRESS = "Email"
export const MEMBER_POSTAL_ADDRESS_WORK = "Address"
export const MEMBER_PROJECT_ROLE = "Role"
export const MEMBER_STATUS = "Status"
export const MEMBER_NAME = "Name"
export const MEMBER_UPLOAD_DIFFERENT_AVATAR = 'Upload a different photo...'
export const MEMBER_FOUNDATIONS_ID = "Foundations ID"
export const MEMBER_FOUNDATIONS_ID_NOT_FOUND = "No Foundations ID for this Email"
export const BUTTON_REMOVE_MEMBER = "Remove From Project"

export const USER_DETAILS_LOADING = "Fetching details, please wait..."

export const TAB_DETAILS = "Details"
export const TAB_HISTORY = "History"
export const TAB_REQUESTS = "Requests"


export const NO_ROLE_SELECTED = "Please select a role..."
export const NO_VALUE_SELECTED = "Please select an option..."

export const FILE_PROMPT = "Choose a file..."
export const FILE_NOT_SELECTED = "No file selected"
export const FILE_READY_TO_UPLOAD = "Ready to upload"
export const FILE_UPLOAD_FAILED = "Upload failed"
export const FILE_UPLOADING = "Upload in progress"
export const FILE_SUCCESSFULLY_UPLOADED = "Successfully uploaded"
export const NO_HASH_YET = "File not uploaded yet"

export const CURRENT_VERSION_ELEMENT = "Current Version"
export const FILE_NAME = 'File Name: '
export const UPLOAD_DATE_TIME = 'Uploaded At: '
export const UPLOADED_BY = 'Uploaded By: '
export const UPLOAD_VERSION = "Version: "
export const NO_CURRENT_VERSION = 'There is no current version of this file.'
export const NO_SIGNATURES = 'There are no signatures for this file'
export const NO_PROOF_AVAILABLE = "Proof still processing"


export const BUTTON_SELF_SIGN_FILE = "Self Sign Document"
export const BUTTON_REQUEST_SIGNATURES = "Request Signatures"
export const BUTTON_UPLOAD_FILE = "Upload File"


export const SIGNATURE_ELEMENT = "Signatures"
export const LINK_TO_PROOF = "Link to Proof"
export const SIGNED_BY = "Signed By: "
export const SIGNED_DATE_TIME = "Signed At: "
export const PROOF = "Proof: "

export const UPLOAD_HISTORY_ELEMENT = "Upload History"
export const NO_PREVIOUS_VERSIONS = "There are no previous version of this file"

export const UPLOAD_IN_PROGESS = "Upload in progress.  Please wait..."
export const UPLOADED_SIZE = "Uploaded: "
export const CLOSE_WINDOW = "Close"

export const PICK_DOCUMENT_SIGNERS = "Pick Document Signatories"
export const SEARCH = "Search"
export const BUTTON_SUBMIT = "Submit"

export const CREATE_CUSTOM_FIELD = "Create Custom Field"
export const FIELD_TITLE = "Field Title"
export const FIELD_DESCRIPTION = "Field Description"
export const FIELD_TYPE = "Field Type"
export const DROP_DOWN_OPTIONS = "Drop Down Options"
export const OPTION_OPENS_TEXT_BOX = "Opens Details Box?"
export const BUTTON_CREATE_FIELD = "Create Field"
export const FIELD_IS_REQUIRED_COMMA_SEPARATED = "(Required, separate options with commas)"
export const DROP_DOWN_OPTIONS_OPENING_TEXT_BOX = "Options that open a \"More Details\" text box:"
export const PLEASE_ENTER_SOME_OPTIONS = "Please enter some options"
export const ERROR_CREATING_CUSTOM_FIELD = "There was an error creating the custom field, please try again.  If the problem persists, please contact technical support."

export const FILE_DETAILS = "File Details"
export const DOWNLOAD = "Download"
export const NO_UPLOAD_NAME = "No name supplied"
export const ERROR_FETCHING_DOWNLOAD_LINK = "There was an error fetching the file, please try again.  If the problem persists, please contact technical support."

export const PROFILE_PAGE_TITLE = "Edit Profile"
export const PROFILE_PAGE_DESCRIPTION = ""

export const CURRENT_PASSWORD = "Current Password"
export const NEW_PASSWORD = "New Password"
export const REPEAT_NEW_PASSWORD = "Repeat New Password"
export const CHANGE_PASSWORD_UNSUCCESSFUL = "Change unsuccessful, please try again"
export const PASSWORDS_DO_NOT_MATCH = "New passwords do not match"

export const USER_UNABLE_TO_FETCH_DETAILS = "Unable to fetch profile details, please try again!"

export const PROJECT_CREATION_IN_PROGRESS = "Project Creation In Progress"
export const PROJECT_CREATION_BILLING_WARNING = "You will only be billed for the project you once you begin uploading data to one of the project tabs (e.g. inception, feasibility, design, etc)"

export const PROJECT_INVITATIONS = "Project Invitations"
export const PROJECT_INVITATION = "Project Invitation"
export const SIGNATURE_REQUESTS = "Signature Requests"
export const SIGNATURE_REQUEST = "Signature Request"
export const BUTTON_ACCEPT = "Accept"
export const BUTTON_REJECT = "Reject"
export const NO_REQUESTS = "You have no requests right now"
export const ERROR_SENDING_SIGNATURE_REQUEST = "There was an error requesting signatures, please try again.  If the problem persists, please contact technical support."
