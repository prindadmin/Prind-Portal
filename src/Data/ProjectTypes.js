import * as Strings from './Strings'

const ProjectTypesList = [
  {
    id: 'CDM2015Project',
    icon: 'building',
    name: Strings.PROJECT_TYPE_CDM2015PROJECT_NAME,
    description: Strings.PROJECT_TYPE_CDM2015PROJECT_DESCRIPTION
  },
  {
    id: 'DHSFProject',
    icon: 'building',
    name: Strings.PROJECT_TYPE_DHSFPROJECT_NAME,
    description: Strings.PROJECT_TYPE_DHSFPROJECT_DESCRIPTION
  }
]

export default ProjectTypesList
