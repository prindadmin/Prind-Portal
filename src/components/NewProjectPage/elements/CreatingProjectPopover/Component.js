import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
