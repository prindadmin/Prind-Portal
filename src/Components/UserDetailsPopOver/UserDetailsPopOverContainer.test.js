import { Suspense } from 'react';
import { Provider, connect } from 'react-redux'
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import ReactGA from 'react-ga';
import { HashRouter as Router } from "react-router-dom";
import * as States from '../../States'
import * as Strings from '../../Data/Strings'
import * as Endpoints from '../../Data/Endpoints'

import Component from './UserDetailsPopOverContainer'

const mockStore = configureStore([]);
var component;

beforeAll(() => {
  ReactGA.initialize('dummy', { testMode: true });
});

beforeEach(() => {
  const store = mockStore({
    auth: {
      isSignedIn: States.AUTH_UNKNOWN,
      isConfirmed: States.AUTH_UNKNOWN,
      hasSignedUp: States.AUTH_UNKNOWN,
      hasSentCode: States.AUTH_UNKNOWN,
      hasChangedPassword: States.AUTH_UNKNOWN,
      passwordResetRequired: States.AUTH_UNKNOWN
    },
    projects: {
      accessibleProjects: {
        projectCreator: [],
        projectRole: []
      },
      chosenProject: {
        projectName: Strings.NO_PROJECT_SELECTED,
        projectId: "",
        projectType: "",
      },
      memberList: [],
      downloadURL: "",
      fileDetails: {},
      fetching: false,
      error: null,
    },
    foundations: {
      fetching: false,
    },
    user: {
      fetching: false,
      details: {},
      history: {},
      projectInvitations: [],
      signatureRequests: [],
      projectS3Token: {},
      userS3Token: {},
      currentRoute: Endpoints.DEFAULTLOGGEDINPAGE,
    },
    members: {
      currentMember: {
        accreditations: [
          {
            issuedDate: "2019-10-01T12:59:12.142216",
            revocationDate: "2019-11-01T12:59:12.142216",
            revocationReason: "expired",
            accreditationID: "constructionBasics",
            accreditationName: "Construction Basics 101",
            subject: "did:fnds:31a24b270fe86d9c595e715854028c319cc75957718861eb66996929eb5c8025",
            issuer: "did:fctr:d85be1f5baa83fa83850d8b58731a7f7c8ba65c33dec107c2e16e0dd65c7bcc7",
            status: "revoked"
          }
        ]
      }
    },
    pageContent: {
      inception: {
        fetching: false,
        fields: [],
        error: null,
      },
      feasibility: {
        fetching: false,
        fields: [],
        error: null,
      },
      design: {
        fetching: false,
        fields: [],
        error: null,
      },
      tender: {
        fetching: false,
        fields: [],
        error: null,
      },
      construction: {
        fetching: false,
        fields: [],
        error: null,
      },
      handover: {
        fetching: false,
        fields: [],
        error: null,
      },
      occupation: {
        fetching: false,
        fields: [],
        error: null,
      },
      refurbishment: {
        fetching: false,
        fields: [],
        error: null,
      },
    }
  });

  const props = {
    memberDetails: {
      username: "aaaa-aaaa-aaaaaaaa-aaaa-aaaa"
    },
    onCancelPopup: jest.fn()
  };

  component = mount(
    <Suspense fallback={<p>error</p>}>
      <Provider store={store}>
        <Router>
          <Component {...props} />
        </Router>
      </Provider>
    </Suspense>
  );
});


it('should render', () => {
  expect(component).toMatchSnapshot();
});
