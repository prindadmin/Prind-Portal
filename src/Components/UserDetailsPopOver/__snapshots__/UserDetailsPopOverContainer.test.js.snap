// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<Suspense
  fallback={
    <p>
      error
    </p>
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <HashRouter>
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <withRouter(Connect(UserDetailsPopOver))
          memberDetails={
            Object {
              "username": "aaaa-aaaa-aaaaaaaa-aaaa-aaaa",
            }
          }
          onCancelPopup={[MockFunction]}
        >
          <Connect(UserDetailsPopOver)
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            memberDetails={
              Object {
                "username": "aaaa-aaaa-aaaaaaaa-aaaa-aaaa",
              }
            }
            onCancelPopup={[MockFunction]}
          >
            <UserDetailsPopOver
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              memberDetails={
                Object {
                  "username": "aaaa-aaaa-aaaaaaaa-aaaa-aaaa",
                }
              }
              members={
                Object {
                  "currentMember": Object {
                    "accreditations": Array [
                      Object {
                        "accreditationID": "constructionBasics",
                        "accreditationName": "Construction Basics 101",
                        "issuedDate": "2019-10-01T12:59:12.142216",
                        "issuer": "did:fctr:d85be1f5baa83fa83850d8b58731a7f7c8ba65c33dec107c2e16e0dd65c7bcc7",
                        "revocationDate": "2019-11-01T12:59:12.142216",
                        "revocationReason": "expired",
                        "status": "revoked",
                        "subject": "did:fnds:31a24b270fe86d9c595e715854028c319cc75957718861eb66996929eb5c8025",
                      },
                    ],
                  },
                }
              }
              onCancelPopup={[MockFunction]}
              tempGetUserAccreditations={[Function]}
            >
              <div
                id="popup-greyer"
                onClick={[Function]}
              >
                <div
                  id="user-details-popover"
                >
                  <div
                    id="popup-box"
                    onClick={[Function]}
                  >
                    <div
                      className="popup-box-header"
                    >
                      <h2>
                        Professional Accreditations
                      </h2>
                      <input
                        className="close-button"
                        id="close-button"
                        onClick={[Function]}
                        type="submit"
                        value="Close"
                      />
                    </div>
                    <div
                      className="accreditation-list-container"
                    >
                      <div
                        className="accreditation-list"
                      >
                        <div
                          className="fill"
                        >
                          <div
                            className="loading-spinner"
                          >
                            <Blueprint3.Spinner
                              intent="danger"
                              size={100}
                            >
                              <div
                                className="bp3-spinner bp3-intent-danger"
                              >
                                <div
                                  className="bp3-spinner-animation"
                                >
                                  <svg
                                    height={100}
                                    strokeWidth="4.00"
                                    viewBox="3.00 3.00 94.00 94.00"
                                    width={100}
                                  >
                                    <path
                                      className="bp3-spinner-track"
                                      d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                                    />
                                    <path
                                      className="bp3-spinner-head"
                                      d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                                      pathLength={280}
                                      strokeDasharray="280 280"
                                      strokeDashoffset={210}
                                    />
                                  </svg>
                                </div>
                              </div>
                            </Blueprint3.Spinner>
                            <p>
                              Fetching Accreditations
                            </p>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </UserDetailsPopOver>
          </Connect(UserDetailsPopOver)>
        </withRouter(Connect(UserDetailsPopOver))>
      </Router>
    </HashRouter>
  </Provider>
</Suspense>
`;
